### PR TITLE
[dev] Singing datasets & ConcatDataset

### PR DESCRIPTION
**Pull request by dependanz**
1. Added support and data split partitions for the singing datasets: `choirset`, `choralsinging`, and `vocalset`.
2. ConcatDataset object to facilitate dataset concatenation.
3. In `promonet.data.pack`, applied `features.detach()` before converting features to an ndarray.
4. Exposed some directory arguments for explicit specification. For example, `promonet.data.augment` now accepts a `--cache_dir` argument, while still defaulting to `promonet.CACHE_DIR`.
5. Removed leftover breakpoint() statements from the code.

### CODE REVIEW

File: promonet/evaluate/metrics.py [hunk 1]
```diff
@@ -306,5 +306,8 @@ def update(self, predicted, target):
 class WER(torchutil.metrics.Average):
     """Word error rate"""
     def update(self, predicted, target):
-        wer = jiwer.wer(target, predicted)
+        try:
```
Comment (CameronChurchwell) [inline promonet/evaluate/metrics.py:309]: What's up with this change? Was something not working with WER? Does it need to be investigated further?

File: promonet/data/preprocess/core.py [hunk 2]
```diff
@@ -9,16 +9,23 @@
...
     for dataset in datasets:
-
+        print(f"preprocessing {dataset}")
         # Get cache directory
         directory = promonet.CACHE_DIR / dataset
 
         # Get text and audio files for this speaker
         audio_files = sorted(list(directory.rglob('*.wav')))
-        audio_files = [file for file in audio_files if '-' in file.stem]
+        # audio_files = [file for file in audio_files if '-' in file.stem]
```
Comment (CameronChurchwell) [inline promonet/data/preprocess/core.py:28]: Also what's up with this change? Presumably this line did _something_ for one or more of the datasets. Or was it a bug?

File: promonet/evaluate/metrics.py (hunk 1 above)
Comment (dependanz) [reply promonet/evaluate/metrics.py:309]: For the singing datasets the sentences were empty, so in the evaluations WER would give division by zero. Hence the bypass `wer=1.0`.

File: promonet/data/preprocess/core.py (hunk 2 above)
Comment (dependanz) [reply promonet/data/preprocess/core.py:28]: I think it was the audio files using the singing dataset didn't have a '-'. I'm not sure if it was a bug, but maybe it was because I didn't follow your cache file naming convention?

**Comment by CameronChurchwell** [comment]
Ok, I'll merge these into dev-staging and tidy some things up before merging into dev.
